### PR TITLE
ref(messaging): Refactored IntegrationBaseParser

### PR DESCRIPTION
**Pull request by iamrajjoshi**
DD: https://www.notion.so/sentry/Task-list-for-Messaging-Vertical-ca4f431d43b84e72b8a91643fe880b66?pvs=4#410553168fb143ef857badd06e50ddbd

Here I move the integration base parser under the integration folder & add logging and metrics to it.

I also updated the original parser as there are 3 integrations in `getsentry` that inherit from it. I will delete the parser here when I update those imports.

### CODE REVIEW

**Comment by iamrajjoshi** [comment]
> 🚨 **Warning:** This pull request contains Frontend and Backend changes!
> 
> It's discouraged to make changes to Sentry's Frontend and Backend in a single pull request. The Frontend and Backend are **not** atomically deployed. If the changes are interdependent of each other, they **must** be separated into two pull requests and be made forward or backwards compatible, such that the Backend or Frontend can be safely deployed independently.
> 
> Have questions? Please ask in the [`#discuss-dev-infra` channel](https://app.slack.com/client/T024ZCV9U/CTJL7358X).

False positive, due to markdown file i updated

File: src/sentry/middleware/integrations/parsers/base.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 from sentry.api.base import ONE_DAY
```
Comment (iamrajjoshi) [inline src/sentry/middleware/integrations/parsers/base.py:15]: probably easier to review this file as it shows the diff of the additions i made. the `integrations/middleware/hybrid_cloud/parser.py` file is a copy of this.

**Review by ykamo001 — APPROVED** [review]

File: src/sentry/middleware/integrations/integration_control_middleware.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ The magic happens in the [`IntegrationControlMiddleware`](src/sentry/middleware/
 
 - If an HTTP request is received to `/extensions/*` (which is the prefix for all our webhooks) it is further inspected. If not, we fall through this middleware.
 - Next, we try to identify the parser from the provider, since these requests follow the pattern `/extensions/provider/webhook-path/`. If no parser is registered, we fall through this middleware
-- If we've found a parser ([`BaseRequestParser`](src/sentry/middleware/integrations/parsers/base.py)), we defer to it for responding to the request, rather than falling through.
+- If we've found a parser ([`BaseRequestParser`](src/sentry/integrations/middleware/hybrid_cloud/parsers.py)), we defer to it for responding to the request, rather than falling through.
```
Comment (markstory) [inline src/sentry/middleware/integrations/integration_control_middleware.md:15]: Should this readme move to the new home as well?

File: src/sentry/middleware/integrations/parsers/base.py [hunk 3]
```diff
@@ -76,8 +82,12 @@ def __init__(self, request: HttpRequest, response_handler: ResponseHandler):
 
     def ensure_control_silo(self):
         if SiloMode.get_current_mode() != SiloMode.CONTROL:
+            metrics.incr(
+                self._METRIC_FAILURE_KEY + ".silo_error",
+                tags={"path": self.request.path, "silo": SiloMode.get_current_mode().value},
+            )
```
Comment (markstory) [inline src/sentry/middleware/integrations/parsers/base.py:88]: You have `sample_rate=1.0` for the success metric on 114-120, but not here or in other metrics that have been moved around. Aligning the sampling rate will be important for when you want to plot these metrics.

**Pull request merged**